packages/blocks/src/_common/utils/query.ts

```diff
@@ -90,46 +90,48 @@ export function getNextBlock(
  *  - paragraph <- when invoked here, the traverse order will be above
  * ```
  *
- * NOTE: this method will skip the `affine:note` and `affine:page` block
+ * NOTE: this method will just return blocks with `content` role
  */
-export function getPreviousBlock(
-  model: BaseBlockModel,
-  map: Record<string, true> = {}
-): BaseBlockModel | null {
-  if (model.id in map) {
-    throw new Error(
-      "Can't get previous block! There's a loop in the block tree!"
-    );
-  }
-  map[model.id] = true;
+export function getPreviousBlock(model: BaseBlockModel): BaseBlockModel | null {
```

Comment:
I think we should use the `getNextBlockCommand` instead of this one.

Reply:
This relates to the refactoring of `rich-text-operations.ts`.